data.js

```diff
@@ -56,3 +56,122 @@ const usersArray = [
     gender: 'female'
   }
 ];
+
+
+// Iteration 1: for...of loop
+const getFirstNames = arr => {
+  const userFirstNames = [];
+  for(let user of arr){
+    let firstName = user.firstName;
+    userFirstNames.push(firstName);
+  }
+
+  console.log(userFirstNames)
+  return userFirstNames; 
+};
+
+getFirstNames(usersArray);
+
+
+
+// Iteration 2: for...of loop and ES6 string literals ${}
+const getFullNames = arr => {
+  const userFullName = [];
+  for(let user of arr){
+    let fullName = `${user.firstName} ${user.lastName}`;
+    userFullName.push(fullName);
+  }
+
+  console.log(userFullName);
+  return userFullName;
+};
+
+getFullNames(usersArray)
+
+
+
+// Iteration 3: ES6 destructuring, for of loop, object literal
+const getUsersCreditDetails = arr => {
+  const usersCreditDetails = [];
+  for(let user of arr) {
+    const {firstName,lastName,balance} = user;
+    let userDetails = {firstName, lastName, balance};
+    usersCreditDetails.push(userDetails);
+  }
+
+  console.log(usersCreditDetails);
+  return usersCreditDetails;
+};
+
+getUsersCreditDetails(usersArray);
+
+
+
+
+// Iteration 4: practice .filter() method and how to return two elements
+const genderView = users => {
+  const femaleUsers = []; 
+  const maleUsers = []; 
+  
+  users.filter(user => {
+    if(user.gender === 'female') {
+      femaleUsers.push(`${user.firstName} ${user.lastName}`);
+    } else {
+      maleUsers.push(`${user.firstName} ${user.lastName}`);
+    }
+  });
+
+  return {
+    femaleUsers,
+    maleUsers
+  }
+};
+
+genderView(usersArray);
+
+
+
+// Bonus - Iteration 5
+const data = genderView(usersArray);
+
+const genderCount = data => {
+  // Your code goes here ...
+  const {femaleUsers, maleUsers} = data;
+  console.log(`Female: ${femaleUsers.length}`);
+  console.log(`Male: ${maleUsers.length}`);
+};
+
+genderCount(data);
+
+
+
+
+// Bonus - Iteration 6
+
+const promo20 = users => {
+  // Your code goes here ...
+  users.map(user => {
+    const userBalance = user.balance.replace('$', "").replace(',', "");
+    const number = parseInt(userBalance);
+
+    if (number > 20000) {
+      console.log(`Dear ${user.firstName}, since your balance is ${user.balance}, you are eligible to apply for this awesome credit card.`);
+    }
+  });
+};
+
+promo20(usersArray);
+
+
+
+
+// Bonus - Iteration 7
+
+const addActive = users => {
+  users.forEach(user => {
+    user.isActive = true;
+  })
```

Comment:
This function immediately is changing the values you are passing onto it.

In functions, unless said otherwise, try to create ahabit that you return a totally new value.
For example
```js
const addActive = users => {
return users.map(user => {
return {
...user, //here you are getting the whole user object copied
isActive: true
}
})
}
```

After this function you get an array of every user active but you get a totally new value.